pkg/cluster/mpi/statusupdate.go

```diff
@@ -105,3 +98,20 @@ func (c statusUpdateComponent) Reconcile(ctx *core.Context) (ctrl.Result, error)
 
 	return ctrl.Result{}, err
 }
+
+func getActivePodCnt(ctx *core.Context, cr *dcv1alpha1.MPICluster) (int, error) {
+	var ep corev1.Endpoints
+	objKey := client.ObjectKey{
+		Name:      serviceName(cr),
+		Namespace: cr.Namespace,
+	}
+
+	if err := ctx.Client.Get(ctx, objKey, &ep); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
Do you know if this makes an API call? Ideally this is only querying a local cache.

Reply:
This is really a copy-paste of Sonny's code from the portion that I have removed. Agree that this logic seems somewhat fragile. I will look at what's going on behind the scene here.